language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -355,6 +372,9 @@ public static String getDefaultValueForType(TypeSymbol bType) {
             case XML:
                 typeString = "xml ``";
                 break;
+            case DECIMAL:
+                typeString = Integer.toString(0);
+                break;
             default:
                 if (typeKind.isIntegerType()) {
```

Comment:
We can add the DECIMAL case here. Both cases have the same logic

Reply:
Will `typeKind.isIntegerType()` be true for decimals?

Reply:
I acutalullay ment addingthe following to line 379
`typeKind.isIntegerType() || typeKind.kind() == TypeDescKind.DECIMAL`. But we can skip the suggestion as it does not have much of an impact

Reply:
Handling it in the case clause will be more efficient IMO. Therefore, marking this as resolved.